diff command bug

# Description
`pachctl diff file <repo@branch1> <repo@branch2> --name-only --no-pager` does not work when the order of the branches is flipped.

# Expected Behavior
The following two commands would run without any runtime errors.
`pachctl diff file <repo@branch1> <repo@branch2> --name-only --no-pager`
`pachctl diff file <repo@branch2> <repo@branch1> --name-only --no-pager`

# Current Behavior
One of the two above commands in `Expected Behavior` will fail, regardless of the images in the branches.

![image](https://user-images.githubusercontent.com/57657083/73986819-6e1f0180-48f3-11ea-9d5f-c9d020770f7f.png)

![image](https://user-images.githubusercontent.com/57657083/73986823-724b1f00-48f3-11ea-954e-12da53975a90.png)


## Steps to Reproduce

(Bash script)

`file1_name="bird.png"`
`file1="https://i.imgur.com/eytrh.jpg"`

`file2_name="cat.png"`
`file2="https://i.imgur.com/cqAklTF.jpg"`

`file3_name="ocean.png"`
`file3="https://i.imgur.com/AuQ9mt9.png"`

`file4_name="mountain.png"`
`file4="https://i.imgur.com/2QeXMqI.png"`

`pachctl delete repo images`
`pachctl create repo images`
`pachctl create branch images@master`
`pachctl create branch images@develop`
`pachctl put file images@master:/$file1_name -f $file1`
`pachctl put file images@develop:/$file2_name -f $file2`

`pachctl diff file --no-pager images@master images@develop`
`pachctl diff file --no-pager images@develop images@master`

## Context

* Operating System: OSX 10.14.6
* Pachyderm version: 1.9.11
* Local Minikube, version 1.17
* VirtualBox 6.1.2
* Git version 2.20.1 (Apple Git-117) 

## Failure Logs

Please include any relevant log snippets or files here.

panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/pachyderm/pachyderm/src/server/pfs/cmds.forEachDiffFile(0xc000628320, 0x2, 0x2, 0xc000628330, 0x2, 0x2, 0xc000687938, 0xc00000e0b8, 0x4762d40)
        src/server/pfs/cmds/cmds.go:1475 +0x38d
github.com/pachyderm/pachyderm/src/server/pfs/cmds.Cmds.func24.1(0x2e07280, 0xc0000d2008, 0x0, 0x0)
        src/server/pfs/cmds/cmds.go:1147 +0x3b2
github.com/pachyderm/pachyderm/src/server/pkg/pager.Page(0x29dbc01, 0x2e07280, 0xc0000d2008, 0xc0002e2a80, 0x0, 0xc000196480)
        src/server/pkg/pager/pager.go:20 +0x300
github.com/pachyderm/pachyderm/src/server/pfs/cmds.Cmds.func24(0xc0001e6fc0, 0x2, 0x4, 0x0, 0x0)
        src/server/pfs/cmds/cmds.go:1127 +0x2c3
github.com/pachyderm/pachyderm/src/server/pkg/cmdutil.RunBoundedArgs.func1(0xc00031d680, 0xc0001e6fc0, 0x2, 0x4)
        src/server/pkg/cmdutil/cobra.go:55 +0x185
github.com/spf13/cobra.(*Command).execute(0xc00031d680, 0xc0001e6f80, 0x4, 0x4, 0xc00031d680, 0xc0001e6f80)
        /home/bryce/go/pkg/mod/github.com/spf13/cobra@v0.0.6-0.20191202130430-b04b5bfc50cb/command.go:833 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc00029a780, 0x0, 0x0, 0xc0004a5e30)
        /home/bryce/go/pkg/mod/github.com/spf13/cobra@v0.0.6-0.20191202130430-b04b5bfc50cb/command.go:917 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /home/bryce/go/pkg/mod/github.com/spf13/cobra@v0.0.6-0.20191202130430-b04b5bfc50cb/command.go:867
main.main.func1(0x0, 0x0)
        src/server/cmd/pachctl/main.go:20 +0x55
main.main()
        src/server/cmd/pachctl/main.go:21 +0xbc

--- a/src/server/pfs/cmds/cmds.go
+++ b/src/server/pfs/cmds/cmds.go
@@ -1503,10 +1503,10 @@ func forEachDiffFile(newFiles, oldFiles []*pfsclient.FileInfo, f func(newFile, o
 		var oFI *pfsclient.FileInfo
 		var nFI *pfsclient.FileInfo
 		switch {
-		case oI == len(oldFiles) || newFiles[nI].File.Path < oldFiles[oI].File.Path:
+		case oI == len(oldFiles) || (nI < len(newFiles) && newFiles[nI].File.Path < oldFiles[oI].File.Path):
 			nFI = newFiles[nI]
 			nI++
-		case nI == len(newFiles) || oldFiles[oI].File.Path < newFiles[nI].File.Path:
+		case nI == len(newFiles) || (oI < len(oldFiles) && oldFiles[oI].File.Path < newFiles[nI].File.Path):
 			oFI = oldFiles[oI]
 			oI++
 		case newFiles[nI].File.Path == oldFiles[oI].File.Path:

